staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -36,53 +33,57 @@ type Route struct {
 	// TODO: add more fields here if needed
 }
 
-func getRoutes(input io.Reader) ([]Route, error) {
-	routes := []Route{}
-	if input == nil {
-		return nil, fmt.Errorf("input is nil")
-	}
-	scanner := bufio.NewReader(input)
-	for {
-		line, err := scanner.ReadString('\n')
-		if err == io.EOF {
-			break
-		}
-		//ignore the headers in the route info
-		if strings.HasPrefix(line, "Iface") {
-			continue
-		}
-		fields := strings.Fields(line)
-		routes = append(routes, Route{})
-		route := &routes[len(routes)-1]
-		route.Interface = fields[0]
-		ip, err := parseIP(fields[1])
-		if err != nil {
-			return nil, err
-		}
-		route.Destination = ip
-		ip, err = parseIP(fields[2])
-		if err != nil {
-			return nil, err
-		}
-		route.Gateway = ip
+const (
+	ipCommand   = "/usr/sbin/ip"
+	grepCommand = "/usr/bin/grep"
+)
+
+var (
+	ipv4_zero = net.IP{0, 0, 0, 0}
+	ipv6_zero = net.ParseIP("::")
+)
+
+var execCommand = exec.Command
+
+// buildRouteCommand forms IP route command(s) to obtain routes and filter
+// on the default routes. Can do IPv4 only, or IPv4 + IPv6.
+func buildRouteCommand(withIPv6 bool) string {
```

Comment:
Thanks for the suggestion. There are 5 functions using the flag (2 just passing it along). I'm wondering if the flag should be added to the existing networkInterface structure, which is already passed around (and still end up passing the flag to the 2 lowest level functions), or if a new struct with methods should be created, as you're suggesting. With the latter, there are three places where it is used, but for different purposes. I'm concerned about having unrelated methods on the structure (buildRouteCommand, findGlobalIP, and buildIPFromHostInterface). Thoughts?